packages/react/src/AsyncAutocomplete/AsyncAutocomplete.tsx

```diff
@@ -49,12 +48,13 @@ export function AsyncAutocomplete<T>(props: AsyncAutocompleteProps<T>): JSX.Elem
     setTimer(undefined);
 
     const value = inputRef.current?.value.trim() || '';
-    if (value === lastValueRef.current) {
-      // Nothing has changed, move on
+    if (value === lastValueRef.current && loadOptions === lastLoadOptionsRef.current) {
```

Comment:
Does `loadOptions` ever change?

Reply:
Yes, this was exposed by the test I added where the target types include a normal resourceType, e.g. `Patient` and an profile of `Patient`, e.g. `USCorePatientProfile`. Switching from one type to the other would cause this to return early even though the search criteria/results should be different, i.e. filtering for patients with the `_profile` query param or not.

Reply:
The same would be true if there were two different profiles of the same resourceType listed which seems like a more likely scenario.